src/services/realtime_monitor.py

```
@@ -220,58 +220,108 @@ def add_data_spec(self, data_spec: str) -> bool:
+                    if code == -114:
+                        logger.debug(f"Spec {data_spec} not subscribed")
+                    elif code == -202:
+                        logger.warning(f"JV-Link busy for {data_spec}, will retry next cycle")
                     else:
-                        logger.error(f"Fetch error for {data_spec}: {e}")
+                        logger.error(f"JVLink error for {data_spec}: {e}")
                         self._add_error(data_spec, str(e))
+                    try:
+                        jvlink.jv_close()
+                    except Exception:
+                        pass
```

Ruff (0.15.10)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)